src/lib/core/utils/visualization.ts

```diff
@@ -130,6 +134,55 @@ export function fixLabelForNumberVariables(
     : label;
 }
 
+/**
+ *
+ * In the abundance app, var ids show up like normal variable values. This function
+ * takes these ids (labels) and returns that variable's display name, if it exists.
+ *
+ * If no variable is found with that id, the original label is returned.
+ */
+export function fixVarIdLabels(
+  labels: string[] = [],
+  variableList: VariableDescriptor[],
+  entities: StudyEntity[]
+): string[] {
+  console.log(labels);
+  const displayNames: string[] = labels.map((label) => {
+    // Label is a variable id. Get entity and var id from variable list.
+    const variableDescriptors = variableList.filter(
+      (varItem) => varItem.variableId === label
+    );
+    const variableDescriptor = variableDescriptors && variableDescriptors[0];
+    const retrievedVariable = findEntityAndVariable(
+      entities,
+      variableDescriptor
+    );
+    return retrievedVariable?.variable.displayName || label;
+  });
+  return displayNames;
```

Comment:
You ought to be able to delegate to `fixVarIdLabel`:
```suggestion
  return labels.map((label) => fixVarIdLabel(label, variableList, entities);
```